pstoremem/addr_book.go

```diff
@@ -96,33 +111,51 @@ func (mab *memoryAddrBook) Close() error {
 // gc garbage collects the in-memory address book.
 func (mab *memoryAddrBook) gc() {
 	now := time.Now()
-	for _, s := range mab.segments {
-		s.Lock()
-		for p, amap := range s.addrs {
+	collect := func(addrs map[peer.ID]map[string]*expiringAddr) []peer.ID {
```

Comment:
Why do we need the separate closure here, and the two-stage deletion? (addresses, then signed records). Can't we just do this as we traverse the segment, using the previous logic?